hdkeychain/extendedkey.go

```diff
@@ -254,7 +254,9 @@ func (k *ExtendedKey) Child(i uint32) (*ExtendedKey, error) {
 		// When the child is a hardened child, the key is known to be a
 		// private key due to the above early return.  Pad it with a
 		// leading zero as required by [BIP32] for deriving the child.
-		copy(data[1:], k.key)
+		// Additionally, right align it if it's shorter than 32 bytes.
+		offset := 33 - len(k.key)
+		copy(data[offset:], k.key)
```

Comment:
Rather than silently changing the behavior of this method, which can cause issues for any wallet that relies on this quirk as is (say when rescanning), I think we should instead _break_ this method, and force callers to choose one or the other. 

In other words: add a new variable to this method, and/or a helper func that assumes a new default value for this variable.

Reply:
renamed `ChildNonStandard` to `DeriveNonStandard` and `Child` to `Derive`.  added a usage note